apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx

```diff
@@ -1,27 +1,167 @@
 import { Heading } from 'components/heading';
-
+import { authorization } from 'authorization';
 import styles from './page-information.module.scss';
+import {
+    PageSummaryDownloadControllerService,
+    PageInformationService,
+    PageInformation as InfoType,
+    PageControllerService,
+    PageHistory
+} from 'apps/page-builder/generated';
+import { download } from '../../../../../../utils/download';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
 
 const PageInformation = () => {
+    const [activeTab, setActiveTab] = useState('Details');
+    const [currentPage, setCurrentPage] = useState(1);
+    const [pageHistory, setPageHistory] = useState<PageHistory[]>([]);
+    const [pageInfo, setPageInfo] = useState<InfoType>({});
+    const { pageId } = useParams();
+    const pageSize = 10;
+    const navigate = useNavigate();
+
+    const fetchPageHistory = async () => {
+        PageControllerService?.getPageHistoryUsingGet?.({
+            authorization: authorization(),
+            id: Number(pageId)
+        }).then((rep) => {
+            setPageHistory(rep);
+        });
+    };
+    const fetchPageInfo = () => {
+        PageInformationService.find({
+            authorization: authorization(),
+            page: Number(pageId)
+        })
+            .then((data: InfoType) => {
+                setPageInfo(data);
+            })
+            .catch((err) => console.error(err));
+    };
+    useEffect(() => {
+        fetchPageInfo();
+        fetchPageHistory();
+    }, []);
+
+    const handleNext = (page: number) => {
+        setCurrentPage(page);
+        fetchPageHistory();
+    };
+    const handleDownloadCSV = async () => {
+        PageSummaryDownloadControllerService.downloadPageMetadataUsingGet({
+            authorization: authorization(),
+            id: Number(pageId)
+        }).then((data) => {
+            const fileType = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+            download({
+                data,
+                fileName: 'PageMetadata.xlsx',
+                fileType
+            });
+        });
+    };
+
+    const handleViewPage = () => {
+        navigate(`/page-builder/pages/page-details/${pageId}`);
+    };
+
+    const renderTabs = (
+        <ul className={styles.tabs}>
+            <li className={activeTab == 'Details' ? styles.active : ''} onClick={() => setActiveTab('Details')}>
+                Details
+            </li>
+            <li className={activeTab == 'History' ? styles.active : ''} onClick={() => setActiveTab('History')}>
+                History
+            </li>
+        </ul>
+    );
+    const renderBlock = (header: string, desc: string) => (
+        <div className={styles.lineBlock}>
+            <div className={styles.detailHeader}>{header}</div>
+            <div className={styles.smallBodyText}>{desc ?? '-'}</div>
+        </div>
+    );
```

Comment:
Suggestion: `desc` is a required input but there is a check to verify it is not `undefined` (`desc ?? '-'`). I would suggest defining  `desc` as:

```typescript
desc?: string
```

Then you can eliminate the `non-null assertion operator` from:  `pageInfo?.eventType?.name!` and other `renderBlock` calls. The use of the `non-null assertion operator` is typically considered a bad practice as this simply overrides the compilers warning without providing a solution.